gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/IamPolicy.java

```diff
@@ -0,0 +1,162 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.resourcemanager;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+import com.google.gcloud.BaseIamPolicy;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Base class for Identity and Access Management (IAM) policies. IAM policies are used to specify
+ * access settings for Cloud Platform resources. A Policy consists of a list of bindings. An binding
+ * assigns a list of identities to a role, where the identities can be user accounts, Google groups,
+ * Google domains, and service accounts. A role is a named list of permissions defined by IAM.
+ *
+ * @see <a href="https://cloud.google.com/iam/reference/rest/v1/Policy">Policy</a>
+ */
+public class IamPolicy extends BaseIamPolicy<String> implements Serializable {
+
+  private static final long serialVersionUID = -5573557282693961850L;
+
+  /**
+   * Builder for an IAM Policy.
+   */
+  protected static class Builder extends BaseBuilder<String, Builder> {
+
+    Builder() {}
+
+    Builder(Map<String, List<Identity>> bindings, String etag, int version) {
+      bindings(bindings).etag(etag).version(version);
+    }
+
+    @Override
+    public IamPolicy build() {
+      return new IamPolicy(this);
+    }
+  }
+
+  private IamPolicy(Builder builder) {
+    super(builder);
+  }
+
+  @Override
+  public int hashCode() {
+    return baseHashCode();
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof IamPolicy)) {
+      return false;
+    }
+    IamPolicy other = (IamPolicy) obj;
+    return Objects.equals(bindings(), other.bindings())
+        && Objects.equals(etag(), other.etag())
+        && Objects.equals(version(), other.version());
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public Builder toBuilder() {
+    return new Builder(bindings(), etag(), version());
+  }
+
+  static String identityToPb(Identity identity) {
+    switch (identity.type()) {
+      case ALL_USERS:
+        return "allUsers";
+      case ALL_AUTHENTICATED_USERS:
+        return "allAuthenticatedUsers";
+      case USER:
+        return "user:" + identity.id();
+      case SERVICE_ACCOUNT:
+        return "serviceAccount:" + identity.id();
+      case GROUP:
+        return "group:" + identity.id();
+      default:
+        return "domain:" + identity.id();
+    }
+  }
+
+  static Identity identityFromPb(String identityStr) {
```

Comment:
Also consider having this method in Identity. Maybe called fromEntity, valueOf, ...

I think using the enum `Type` would be nicer. Also, we should consider that a ':' may
be part of the value/id (is it documented otherwise)?

```java
int idx = identityStr.indexOf(':');
Type type = Type.valueOf(idx < 0 ? identityStr : identityStr.substring(0, idx));
switch (type) {
  case ALL_USERS:
    return Identity.allUsers();
  ....
  case USER:
    return Identity.user(identityStr.substring(idx + 1));
  ...
}
```